dt/vector2.cpp

```diff
@@ -20,12 +20,9 @@ template<typename T>
 T
 Vector2<T>::dist(const Vector2<T> &v) const
 {
-	if constexpr (std::is_same_v<T, float>) {
-		return hypotf(x - v.x, y - v.y);
-	} else if constexpr (std::is_same_v<T, double>) {
-		return hypot(x - v.x, y - v.y);
-	}
-	static_assert(true, "Must be floating-point type");
+	static_assert(std::is_floating_point<T>::value,
+		"Must be floating-point type");
+	return std::hypot(x - v.x, y - v.y);
```

Comment:
Same goes as the comment for `almost_equal`.

Reply:
Here we can't use the same `enable_if` magic as above since it's a method inside a template class. I replaced it with two separate (useless :wink:) template specializations, so the compiler will call the corresponding method depending on the class template type.